### PR TITLE
Add KPP Langmuir enhancement factor using WAVEWATCHIII Stokes drift

### PR DESCRIPTION
**Pull request by sbrus89**
Up to 6 wavenumber-partitioned surface Stokes drift fields are coupled to MPAS-O and used to reconstruct the Stokes drift profile and compute surface-layer averages. The surface layer average of the Stokes drift is used to compute the Langmuir number and KPP enhancement factor from Van Roekel et al. (2012).

(This is a duplicate of #5, which was mistakenly marked as closed after I push the wrong branch to OceanNGD/add-ww3)

### CODE REVIEW

**Comment by sbrus89** [comment]
@vanroekel, I'd appreciate you looking this one over pretty carefully when you get a chance.

**Comment by sbrus89** [comment]
The relevant MPAS-O changes are here:

https://github.com/sbrus89/MPAS-Model/compare/6af0a6e55f4c9b97d098d6fedda5b46182299caa..2f10b1d0af03a7d08e2cd95f1926a274b5fde2a6

**Comment by vanroekel** [comment]
@sbrus89 thanks for the PR already.  Happy to see this happen!  I’m looking into this now and have a few high level requests/questions

1) Could you post the reference you follow to compute Us?  Is it the Brevik et al paper?
2) Could you plot a few sample stokes drift profiles and the resulting diffusivity?
3) why only allow partitions of 3,4,6 and not 5?
4) I would advocate that we keep the computation of stokes drift wholly separate from KPP.  All of the options about configurations the calculation I would not name them with reference to KPP

I’ll keep digging in.

File: components/mpas-ocean/bld/build-namelist [hunk 1]
```diff
@@ -644,11 +644,14 @@ add_default($nl, 'config_cvmix_kpp_nonlocal_with_implicit_mix');
 if ($ocn_wave eq 'true' ) {
 	add_default($nl, 'config_cvmix_kpp_use_theory_wave', 'val'=>'.false.');
 	add_default($nl, 'config_cvmix_kpp_use_active_wave', 'val'=>'.true.');
+	add_default($nl, 'config_cvmix_kpp_n_wavenumber_partitions');
```
Comment (vanroekel) [inline components/mpas-ocean/bld/build-namelist:645]: I would suggest separating the wavenumber partition from KPP

File: components/ww3/src/cpl/ww3_cpl_indices.f90 [hunk 2]
```diff
@@ -18,9 +18,20 @@ module ww3_cpl_indices
   integer :: index_x2w_So_bldepth     
   integer :: index_x2w_So_ssh
 
-  integer :: index_w2x_Sw_ustokes
-  integer :: index_w2x_Sw_vstokes
-  integer :: index_w2x_Sw_hstokes
+  integer :: index_w2x_Sw_ustokes_wavenumber_1
```
Comment (vanroekel) [inline components/ww3/src/cpl/ww3_cpl_indices.f90:23]: mostly a curiosity that could have been anywhere, what does `Sw` stand for throughout the code?

Comment (philipwjones) [reply components/ww3/src/cpl/ww3_cpl_indices.f90:23]: In the coupler or coupled context, the convention is that S is for a state variable (F for flux) and the small letter is the component (w for wave). And stuff that happens in the coupler is x, so w2x is wave to coupler.

Comment (vanroekel) [reply components/ww3/src/cpl/ww3_cpl_indices.f90:23]: thanks @philipwjones good to know!

**Comment by sbrus89** [comment]
Thanks for the comments @vanroekel.

>     1. Could you post the reference you follow to compute Us?  Is it the Brevik et al paper?

What I implemented here isn't the Brevik et al. approach. I ended going with the wavenumber-partiioned approach described in [Li et al. 2019](https://agupubs.onlinelibrary.wiley.com/doi/full/10.1029/2019MS001810I). This uses the surface Stokes drift evaluated at different wavenumbers, which can then be used to reconstruct the full profile (see Eqn. 9):
Screen Shot 2021-06-08 at 2 28 45 PM


>     2. Could you plot a few sample stokes drift profiles and the resulting diffusivity?

I'm still debugging right now, but I will do this once things are fully working.

>     3. why only allow partitions of 3,4,6 and not 5?

I reached out to Brandon Reichl and he graciously shared the wavenumbers he has been using. The values he gave me were for 3, 4, and 6.

>     4. I would advocate that we keep the computation of stokes drift wholly separate from KPP.  All of the options about configurations the calculation I would not name them with reference to KPP

Good suggestion; I agree. I will make this change.

**Pull request closed without merging**